packages/babel-generator/src/generators/statements.js

```diff
@@ -123,10 +123,14 @@ function buildLabelStatement(prefix, key = "label") {
     const label = node[key];
     if (label) {
       this.space();
-
-      const terminatorState = this.startTerminatorless();
-      this.print(label, node);
-      this.endTerminatorless(terminatorState);
+      if (prefix == "break" || prefix == "continue") {
+        // see https://git.io/vQ7uG
+        this.print(label, node);
```

Comment:
@loganfsmyth @jridgewell what do you think about converting single line comments after `break` or `continue` to multi line instead of trying to move them above the `break` or `continue`?

not that I know an easy way, but I could look for ideas .. 

e.g from my comment https://github.com/babel/babel/issues/5742#issuecomment-315352527

``` js
switch (_ret) {case "continue":continue;case "break":break /* istanbul ignore next */_loop2;}
```
that seems like valid code, and nothing broke too